app/src/main/java/org/mozilla/fenix/settings/logins/LoginDetailFragment.kt

```diff
@@ -0,0 +1,243 @@
+            loginDetailView.update(it)
+            login = savedLoginsStore.state.currentItem
+//            showToolbar(savedLoginsStore.state.currentItem?.origin ?: "")
+            setUpCopyButtons()
+            showToolbar(savedLoginsStore.state.currentItem?.origin ?: "")
+            setUpPasswordReveal()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
+        super.onCreate(savedInstanceState)
+    }
+
+    private fun setUpPasswordReveal() {
+        passwordInfoText.inputType =
+            InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+        revealPasswordButton.setOnClickListener {
+            togglePasswordReveal()
+        }
+    }
+
+    private fun setUpCopyButtons() {
+        val copyInfoArgs = listOf(
+            Pair(copyWebAddress, login?.origin),
+            Pair(copyUsername, login?.username),
+            Pair(copyPassword, login?.password)
+        )
+
+        for (copyButtons in copyInfoArgs) {
+            copyButtons.first.setOnClickListener(
+                CopyButtonListener(
+                    copyButtons.second,
+                    R.string.login_copied
+                )
+            )
+        }
+    }
+
+    // TODO: Move interactions with the component's password storage into a separate datastore
+    private fun fetchLoginDetails() {
+        var deferredLogin: Deferred<List<Login>>? = null
+        val fetchLoginJob = lifecycleScope.launch(Dispatchers.IO) {
+            deferredLogin = async {
+                requireContext().components.core.passwordsStorage.list()
+            }
+            val fetchedLoginList = deferredLogin?.await()
+
+            fetchedLoginList?.let {
+                withContext(Dispatchers.Main) {
+                    val login = fetchedLoginList.filter {
+                        it.guid == args.savedLoginId
+                    }.first()
+                    savedLoginsStore.dispatch(
+                        LoginsAction.UpdateCurrentLogin(login.mapToSavedLogin())
+                    )
+                }
+            }
+        }
+        fetchLoginJob.invokeOnCompletion {
+            if (it is CancellationException) {
+                deferredLogin?.cancel()
+            }
+        }
+
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.login_options_menu, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean = when (item.itemId) {
+        R.id.delete_login_button -> {
+            displayDeleteLoginDialog()
+            true
+        }
+        R.id.edit_login_button -> {
+            editLogin()
+            true
+        }
+        else -> false
+    }
+
+    private fun editLogin() {
+        val directions =
+            LoginDetailFragmentDirections
+                .actionLoginDetailFragmentToEditLoginFragment(login!!)
+        requireView().findNavController().navigate(directions)
```

Comment:
nit - I don't think you need the `requireView` here, `findNavController` should work on the fragment